scripts/commit_classifier.py

```diff
@@ -146,6 +152,29 @@ def __init__(self, model_name, repo_dir, git_repo_dir, method_defect_predictor_d
                 "8cc47f47ffb686a29324435a0151b5fabd37f865",
             )
 
+        if revision is not None:
+            assert phabricator_deployment is None
+            assert diff_id is None
+            self.revision = revision
+
+        self.update_commit_db()
+
+        if diff_id is not None:
+            assert phabricator_deployment is not None
+            assert revision is None
+
+            with hglib.open(self.repo_dir) as hg:
+                if phabricator_deployment is not None:
+                    self.apply_phab(hg, phabricator_deployment, diff_id)
+
+                    self.revision = hg.log(revrange="not public()")[0].node.decode(
+                        "utf-8"
+                    )
+
+                    # model will get downloaded only if patch is applied
+                    self.model = download_and_load_model(model_name)
```

Comment:
The model needs to be downloaded also if revision is not None and diff_id is None, so it should be outside the `         if diff_id is not None:`

Reply:
If the patch fails to apply, we'll stop here anyway, so we won't reach the download code.

Reply:
But here, we don't want to download the model, if we don't apply the patch right?
https://github.com/mozilla/bugbug/pull/1697#discussion_r455419168

Reply:
Oh okay, got it.
Will fix it right away, thank you